### PR TITLE
Finished Onboarding Task 0

### PR DESCRIPTION
**Pull request by gabriel-jt**
## In this PR

Followed along the tutorial linked in Notion onboarding and remade the Recipe app using Typescript.

### 🖋  What I learned

- Basic structure of React projects and how different parts of the project interact with each other (functional components/ front end and API requests/backend)
- How to create reusable front end components with functional components
- Differences between JavaScript and TypeScript
- Learned about useStates and useEffect functions to update components and pages.
- Learned about type basics in TypeScript and how to use them.

### 🤳 Screenshots (if applicable)

Screen Shot 2021-03-06 at 11 11 50 PM

### 🐞 Questions

When do we usually use the type `any` for a variable?

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  useEffect( () => {
+    getRecipes();
+  }, [query]);
+
+  const getRecipes = async (): Promise<void> => {
+    const response = await fetch(`https://api.edamam.com/search?q=${query}&app_id=${APP_ID}&app_key=${APP_KEY}`);
+    const data = await response.json();
+    setRecipes(data.hits);
+    console.log(data.hits);
```
Comment (0xFranc0) [inline src/App.tsx:22]: Usually your PL will mention to remove `console.log` statements in your commits or your linter will catch your console.log statements before you commit using "pre-commit" hooks, which you can think of as scripts that run and check your code before it's committed.  This is totally fine for our onboarding activity 👍

File: src/Recipe.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    return(
+        <div className={style.recipe}>
+            <h1>{title}</h1>
+            <ol>
+                {ingredients.map(ingredient => (
+                    <li>{ingredient.text}</li>
+                ))}
+            </ol>
+            <p>{calories}</p>
+            <img className={style.image} src={image} alt="" />
```
Comment (0xFranc0) [inline src/Recipe.js:13]: For images, your PL might ask you to provide an `alt` description to help make your project's application as accessible as possible.  

A possible addition here would be

```
<img className={style.image} src={image} alt={`${ingredient.text} ingredient`} />

```

File: src/components/RecipeCard.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    return(
+        <div className={style.recipe}>
+        <h1>{title}</h1>
+        <ol>
+            {ingredients.map((ingredient : any) => (
+                <li>{ingredient.text}</li>
+            ))}
+        </ol>
+        <p>{calories}</p>
+        <img className={style.image} src={image} alt="" />
```
Comment (0xFranc0) [inline src/components/RecipeCard.tsx:26]: (Same as the comment above in regards to img alts)

File: src/components/RecipeCard.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import style from "./recipecard.module.css";
+
+interface RecipeProps {
```
Comment (0xFranc0) [inline src/components/RecipeCard.tsx:4]: Awesome use of an interface!

**Review by 0xFranc0 — APPROVED** [review]
LGTM! (looks good to me!) Gabe!!! 🚀

Awesome job! Not only did you complete the recipe app but you also took it as an opportunity to learn TypeScript in the context of react! ++

This PR is officially approved, so feel free to merge this in 👍 

I left some comments that you might get from your PL or teammates -- these do not need to be addressed -- this is more to provide context when you go into your project work.

On that note -- it looks like you learned a lot! Very awesome!

Questions
> When do we usually use the type any for a variable?
Usually we would avoid using the `any` type for a variable, but there are cases where you might be working with an external api/third party service for the first time and do not know what type of data they are providing, so in those cases, when assigning the payload from those apis and assigning it to a variable `any` would be a good use case.

Great question!

**Comment by gabriel-jt** [comment]
Thanks for the feedback, Franco! Will definitely keep this in mind as I complete sprint tasks!

**Pull request merged**